gh-93243: Make smtpd private before porting its users (#93246)
python/cpython: Misc/NEWS.d/next/Library/2022-05-26-08-41-34.gh-issue-93243.uw6x5z.rst

> The :mod:`smtpd` dead battery was removed as slated by :pep:`594`.

Reviewer: Slightly more neutral wording, feel free to keep "dead battery" if it tickles your fancy!
```suggestion
The :mod:`smtpd` module was removed per the schedule in :pep:`594`.
```
